#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: fix ioperm build error on arm

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

--- sysdeps/unix/sysv/linux/arm/ioperm.c	2003-02-20 22:22:35.000000000 +0000
+++ sysdeps/unix/sysv/linux/arm/ioperm.c	2003-10-15 17:24:45.000000000 +0000
@@ -44,6 +44,7 @@
 #include <sys/types.h>
 #include <sys/mman.h>
 
+#include <linux/input.h>
 #include <asm/page.h>
 #include <sys/sysctl.h>
 
